src/main/java/seedu/manager/parser/Parser.java

```diff
@@ -992,20 +1002,26 @@ private Command parseSortCommand(String input, String[] commandParts) throws Inv
      */
     private Command parseFilterCommand(String input, String[] commandParts) throws InvalidCommandException {
         assert commandParts[0].equalsIgnoreCase(FilterCommand.COMMAND_WORD);
-        String[] inputParts = input.split("(-e|-d|-t|-x|-u)");
-        if (inputParts.length < 2) {
-            throw new InvalidCommandException(INVALID_FILTER_MESSAGE);
-        }
 
-        if (inputParts[1].isBlank()) {
-            throw new InvalidCommandException(EMPTY_INPUT_MESSAGE);
-        }
+        checkForDuplicateFlags(input, FILTER_FLAG_REGEX);
+
+        Pattern pattern = Pattern.compile(FILTER_REGEX);
+        Matcher matcher = pattern.matcher(input);
+
+        if (matcher.matches()) {
 
-        Set<String> validFlags = Set.of(EVENT_FLAG, "-d", "-t", "-x", "-u");
-        if (validFlags.contains(commandParts[1].trim().toLowerCase())) {
-            return new FilterCommand(commandParts[1].trim().toLowerCase(), inputParts[1].trim());
+            if (matcher.group(1).isBlank()) {
```

Comment:
matcher.group(1) should be the filter description keywords. There isn't a matcher.group(2)

Reply:
i used group one as the flag cause got multiple types of flag that i need to pass to the filter command

Reply:
As in I do not think matcher.group(2) exists? Like there should not be any value inside it

Reply:
ok my bad